shopping_mall/backend/scripts/evaluate_rag.py

```
@@ -0,0 +1,750 @@
+    _type_order = ["colloquial", "synonym", "multi_intent", "direct",
+                   "exact_number", "acronym", "sparse_match"]
+    all_types = _type_order + sorted(
+        t for t in set(list(base.per_type.keys()) + list(imp.per_type.keys()))
+        if t not in _type_order
+    )
+    for t in all_types:
+        if t not in base.per_type and t not in imp.per_type:
+            continue
+        b = base.per_type.get(t, {})
+        i = imp.per_type.get(t, {})
+        b_h1 = b.get("hit1_rate", 0); i_h1 = i.get("hit1_rate", 0)
```

Ruff (0.15.12)

[error] Multiple statements on one line (semicolon)

(E702)